native/cocos/platform/android/java/src/com/cocos/lib/CocosDownloader.java

```diff
@@ -136,43 +134,27 @@ public void run() {
                 Call task = null;
 
                 do {
-                    if (path.length() > 0) {
-                        try {
-                            URI uri = new URI(url);
-                            domain = uri.getHost();
-                        } catch (URISyntaxException e) {
-                            e.printStackTrace();
-                            break;
-                        } catch (NullPointerException e) {
-                            e.printStackTrace();
-                            break;
-                        }
-
+                    if (pathWithUrlHash.length() > 0) {
                         // file task
-                        tempFile = new File(path + downloader._tempFileNameSuffix);
+                        tempFile = new File(tempFilePath);
                         if (tempFile.isDirectory()) break;
 
                         File parent = tempFile.getParentFile();
                         if (!parent.isDirectory() && !parent.mkdirs()) break;
 
-                        finalFile = new File(path);
+                        finalFile = new File(pathWithUrlHash);
                         if (finalFile.isDirectory()) break;
                         long fileLen = tempFile.length();
-
-                        host = domain.startsWith("www.") ? domain.substring(4) : domain;
                         if (fileLen > 0) {
-                            if (_resumingSupport.containsKey(host) && _resumingSupport.get(host)) {
                                 downloadStart = fileLen;
-                            } else {
-                                // Remove previous downloaded context
-                                try {
-                                    PrintWriter writer = new PrintWriter(tempFile);
-                                    writer.print("");
-                                    writer.close();
-                                }
-                                // Not found then nothing to do
-                                catch (FileNotFoundException e) {
-                                }
+                        } else {
+                            try {
+                                PrintWriter writer = new PrintWriter(tempFile);
+                                writer.print("");
+                                writer.close();
+                            }
+                            // Not found then nothing to do
+                            catch (FileNotFoundException e) {
```

Comment:
No need to empty the tempFile, since it's length 0 ?

Reply:
Yes. remove it